evap/results/tests/test_views.py

```diff
@@ -58,6 +58,36 @@ def test_single_result_course(self):
         response = self.app.get(url, user=user)
         self.assertEqual(response.status_code, 200, 'url "{}" failed with user "{}"'.format(self.url, user))
 
+    def test_private_course(self):
+        student = UserProfile.objects.get(username="student")
+        contributor = UserProfile.objects.get(username="contributor")
+        responsible = UserProfile.objects.get(username="responsible")
+        test1 = mommy.make(UserProfile, username="test1")
+        test2 = mommy.make(UserProfile, username="test2")
+        mommy.make(UserProfile, username="random")
+        private_course = mommy.make(Course, state='published', is_private=True, semester=self.semester, participants=[student, test1, test2], voters=[test1, test2])
+        mommy.make(Contribution, course=private_course, contributor=responsible, can_edit=True, responsible=True, comment_visibility=Contribution.ALL_COMMENTS)
+        mommy.make(Contribution, course=private_course, contributor=contributor, can_edit=True)
+
+        url = '/results/semester/%s' % (self.semester.id)
+        page = self.app.get(url, user='random')
+        self.assertNotIn(private_course.name, page)
+        page = self.app.get(url, user='student')
+        self.assertIn(private_course.name, page)
```

Comment:
the test fails here, because no courses are listed on the results page. but i don't know why.